### ISSUE TITLE
mocking posthog

### ISSUE THREAD

**Issue by edublancas** [ploomber/core#3]
we are missing the fixture that prevents posthog from hitting the API:

https://github.com/ploomber/ploomber/blob/54c26e1885c019f4a954d3f4b37ed136cd72a2fc/tests/conftest.py#L63

we should add it to the conftest:

https://github.com/ploomber/core/blob/main/tests/conftest.py

but we should also keep the copy in ploomber

cc @Wxl19980214

**Comment by Wxl19980214** [comment]
PR up, also made posthog api key as a parameter to pass in when call log_api. 

Bump core to newest version and then I can test in sooperviosr and ploomber

**Comment by edublancas** [comment]
alright, overall, it looks good. I saw you added the key argument to the log_api since we'll now have multiple of them. I'd recommend another approach so we don't have to use the API key repeatedly.


```python
class Telemetry:
    def __init__(self, api_key):
        self._api_key = api_key

    def log_api(self, ...):
        # we can use self._api_key here
        ...

# then to use it


telemetry = Telemetry(api_key)

@telemetry.log_api
def some_function():
    ...

```


I think the `log_api` will still work once you move it to the class (just add self as the first argument). but if you encounter issues you might need to dig a bit into [python decorators](https://realpython.com/primer-on-python-decorators/)

**Comment by edublancas** [comment]
hey @Wxl19980214 did you get a change to see my message above? If there's confusion about the task. ping me in the morning and we can schedule a quick call

**Comment by Wxl19980214** [comment]
Sorry I think I missed the message and I have been working on different things. I will take a look tmr

**Comment by Wxl19980214** [comment]
I think the problem now is we cant do this because the naming conflict:
```
telemetry = Telemetry(api_key)

@telemetry.log_api
def some_function():
    ...
```

However, this indeed pass the unit tests so I assume it works as well:
```
telemetry_instance = Telemetry(api_key)

@telemetry_instance.log_call('action')
def some_function():
    ...

```

But I don't know if this is suits naming conventions tho.

**Comment by edublancas** [comment]
that's fine, we can dom

```python
from module.telemetry import Telemetry

telemetry = Telemetry()
```

**Comment by edublancas** [comment]
fixed by #5
